### PR TITLE
Docker

### LINKED ISSUE #23: Create container with dependencies
## The problem

Thinking in situations similar to Graham / ComputeCanada were there is no internet access to download dependencies. Could make use of a container in that situation.

TODO:

* Install software packages:
  - [x] ANTs (need to double check version(s) used and verify this works for workflow
    * `neuroglia-core` is used for one version of ANTs, while ANTs container used for another (due to bug with registration)
  - [x] Freesurfer
  - [x] Mrtrix
  - [x] Labelmerge (via `pip` / `poetry`) 

## Environment

N/A

### PR DESCRIPTION
**Pull request by kaitj**
## Proposed changes

This updates the Docker file so everything is contained within the singularity image (allowing it to be directly run from the container), with the container itself being around 6GB. 

Should be ready for a review of the code, but needs to be tested on actual data prior to merging. Would also like to **hold off on merging this** so that a `0.1.0` version can first be released (basically everything that was used as part of the OHBM abstract) Also planning to, in a future PR, update all of the containers to make use of this new container.

Resolves #23 and #41.

One thing I would like to do here is to merge this, and before creating the next release update the config with the next docker version so that the workflow actually uses the self-contained with updated dependencies.

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [ ] Changes have been tested to ensure that fix is effective or that a feature works.
- [ ] Changes pass the unit tests
- [ ] Code has been run through the `poe quality` task
- [x] I have included necessary documentation or comments (as necessary)
- [ ] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

### CODE REVIEW

**Comment by kaitj** [comment]
Would propose holding this off until #44 and #43 are merged in, and this is rebased following the name change. This would be more of a `v0.1.1` release.

File: pyproject.toml
Comment (tkkuehn) [inline pyproject.toml:13]: I don't _think_ this should be necessary but even if not can keep it if you want

**Review by tkkuehn — APPROVED** [review]
This basically looks good to me, it may be possible to slim down the docker image a bit by only copying the outputs from each build stage though (along the same lines as my comment [here](https://github.com/khanlab/labelmerge/pull/38#discussion_r1103243284)). The general approach I would use is:

- Have a `requirements` stage where you install all the apt requirements you'll ultimately need for the runtime stage.
- For each tool stage, refer to the `requirements` stage, install any build-time requirements, and build the tool.
- For the `runtime` stage, refer to the `requirements` stage, only copy the built artifacts you need from the tool stages, and install the wheel like you're doing here.

For reference [the Dockerfile for the autobids portal](https://github.com/khanlab/autobids-newstudy-app/blob/main/Dockerfile) basically uses this approach, but only installs Apptainer.

All that said, if you don't want to go to the effort which is potentially totally reasonable, this can be merged as-is.

**Comment by kaitj** [comment]
@tkkuehn Mind taking another look at this when you get a chance? The `Dockerfile` was updated and there were some workflow updates as a result as well.

**Review by tkkuehn — APPROVED** [review]

**Pull request merged**
